cmd/integration/commands/export_db_data.go

```diff
@@ -0,0 +1,584 @@
+			if blockNum < fromNum {
+				continue
+			}
+
+			if blockNum >= toNum {
+				break
+			}
+
+			var blockHash libcommon.Hash
+			copy(blockHash[8:], k)
+
+			td := new(big.Int)
+			if err := rlp.Decode(bytes.NewReader(v), td); err != nil {
+				logger.Error(err.Error())
+				return
+			}
+
+			record := []string{"DB", strconv.FormatUint(blockNum, 10), blockHash.Hex(), td.String()}
+			if err := csvWriter.Write(record); err != nil {
+				logger.Error(err.Error())
+				return
+			}
+		}
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		csvWriter.Flush()
+		if err = csvWriter.Error(); err != nil {
+			logger.Error(err.Error())
+			return
+		}
+	},
+}
+
+var cmdExportHeimdallEvents = &cobra.Command{
+	Use:   "export_heimdall_events",
+	Short: "",
+	Run: func(cmd *cobra.Command, args []string) {
+		logger := debug.SetupCobra(cmd, "integration")
+		label := kv.ChainDB
+		db, err := openDB(dbCfg(label, path.Join(datadirCli, label.String())), true, logger)
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		defer db.Close()
+		tx, err := db.BeginRo(cmd.Context())
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		defer tx.Rollback()
+		snapDir := path.Join(datadirCli, "snapshots")
+		logger.Info("snapshot dir info", "dir", snapDir)
+
+		allSnapshots := freezeblocks.NewRoSnapshots(ethconfig.BlocksFreezing{}, snapDir, 0, logger)
+		if err := allSnapshots.ReopenFolder(); err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		allBorSnapshots := freezeblocks.NewBorRoSnapshots(ethconfig.BlocksFreezing{}, snapDir, 0, logger)
+		if err := allBorSnapshots.ReopenFolder(); err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		eventSegments := allBorSnapshots.ViewType(snaptype.BorEvents).VisibleSegments
+		blockReader := freezeblocks.NewBlockReader(allSnapshots, allBorSnapshots)
+		lastFrozenEventId := blockReader.LastFrozenEventId()
+		iterateSnapshots := lastFrozenEventId > 0 && fromNum <= lastFrozenEventId
+		iterateDb := true
+
+		out, err := os.Create(outputCsvFile)
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		csvWriter := csv.NewWriter(out)
+		err = csvWriter.Write([]string{"source", "id", "time_unix_sec", "date_time_utc", "tx_hash", "log_index"})
+		if err != nil {
+			logger.Error(err.Error())
+			return
+		}
+
+		writeEventRow := func(source string, event *heimdall.EventRecordWithTime) error {
+			return csvWriter.Write([]string{
+				source,
+				strconv.FormatUint(event.ID, 10),
+				strconv.FormatInt(event.Time.Unix(), 10),
+				event.Time.UTC().Format(time.RFC3339),
+				event.TxHash.String(),
+				strconv.FormatUint(event.LogIndex, 10),
+			})
+		}
+
+	snapshotLoop:
+		for i := 0; iterateSnapshots && i < len(eventSegments); i++ {
+			seg := eventSegments[i]
+			getter := seg.MakeGetter()
+			for getter.HasNext() {
+				var buf []byte
+				buf, _ = getter.Next(buf)
+
+				eventBytes := rlp.RawValue(libcommon.Copy(buf[length.Hash+length.BlockNum+8:]))
```

Comment:
you dont use `eventBytes` outside of loop iteration, then no reason to `common.Copy`